addons/binding/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartImpl.java

```diff
@@ -342,9 +337,10 @@ public void sendCommand(Device device, GardenaSmartCommandName commandName, Obje
         }
 
         if (command != null) {
+            stopRefreshThread();
```

Comment:
What was wrong with your previous "intermediate" refresh?

> So i have to cancel the refresh thread

You currently do this with the `true` parameter, which means that it is interrupted, even if the job is executed at that moment. I'd think you should give any existing job the chance to finish and wait for its termination first.